docs/rfcs/016-asset-metadata.md

```diff
@@ -0,0 +1,95 @@
+
+The above values can then be used to drive generation of improved [single-item manifest](https://github.com/dlcs/protagonist/issues/488) and clearing up [no-longer required delivery artifacts](https://github.com/dlcs/protagonist/issues/430).
+
+### Database
+
+#### Table Name
+
+Considered names and reason for choosing or not:
+* `AssetApplicationMetadata` - chosen name as doesn't add restriction to what is being stored but `Application` in name highlights that this is for internal metadata only, not replacements for string1, string2 etc.
+* `ImageMetadata` - while suitable this is vague and opens up the table to be a dumping ground for any data.
+* `DeliveryChannelMetadata` - originally considered name, this would point to having both an `assetId` and `channel` record per row but some values may be relevant for multiple channels (namely image + thumbs). We may also want to store some data that is not delivery-channel specified (e.g. checksum).
+  * If we opted for this table we could consider adding a new column to the [`ImageDeliveryChannel`](014-delivery-channels-database.md) table, as this would store a row per Asset/Delivery channel.
+* `AssetDeliveryMetadata` - similar to above - if we are storing checksum etc this isn't asset delivery (ie Orchestrator/Thumbs) specific.
+
+#### Schema
+
+The suggested schema for the table should be flexible.
+
+| Column        | Type        | Description                         |
+| ------------- | ----------- | ----------------------------------- |
+| AssetId       | text        | AssetId this is for                 |
+| MetadataType  | text        | Identifier for the type of metadata |
+| MetadataValue | jsonb       | JSON object of values for type      |
+| {audit-cols}  | timestamptz | Created/updated dates               |
+
+* `AssetId` - this maintains link back to asset. Storing Id only is fine, no need to store separate `customer` or `space` as lookup will only be by Id.
+* `MetadataType` is the 'key' used to look up relevant metadata - these values won't link to anything in the database but a known list of values will be maintained and used by the application code.
+* `MetadataValue` is a `jsonb` value storing relevant data as JSON. In most cases I think we would always want to read the entire object but it could be useful to have efficient querying afforded by `jsonb` (e.g. to get `"o"`pen thumbs only). This querying has support in [npgsql](https://www.npgsql.org/efcore/mapping/json.html).
+* `AssetId` and `MetadataType` would be composite key.
+
+#### Sample Data
+
+Example data for an asset could be:
+
+| AssetId | MetadataType | MetadataValue                                                     |
+| ------- | ------------ | ----------------------------------------------------------------- |
+| 1/2/foo | ThumbSizes   | `{"o": [[200,127],[100,64]],"a": [[1024,651], [400,254]]}`        |

```

Comment:
Not sure of the details but might there be a reason to store the thumb sizes differently from s3 here? e.g., some sort of query by size. This layout suits the needs of thumb serving but may not be the right shape for other uses (although it might be).